### PR TITLE
Delete utils module

### PR DESCRIPTION
**Pull request by vegaro**
Based off #1097 

Before:
image

After:
image

Note:

- `JSONObjectHelper` and `DateHelper` live in `public` because they are used in that module and couldn't be moved to `common`

### CODE REVIEW

File: public/src/main/java/com/revenuecat/purchases/utils/DateHelper.kt
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/utils/DateHelper.kt:1]: Note this one and `JSONObjectParceler` live in `public` because they are used there

File: public/src/main/java/com/revenuecat/purchases/utils/DateHelperDupe.kt
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/utils/DateHelperDupe.kt:1]: Had to made some duplicates in this package because moving those files into the `public` module would expose them to everyone

File: public/src/main/java/com/revenuecat/purchases/utils/Iso8601UtilsDupe.kt
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/utils/Iso8601UtilsDupe.kt:1]: Converted this file to Kotlin using Android Studio so it can be made internal and we can hide it from developers

File: public/src/main/java/com/revenuecat/purchases/utils/DateHelperDupe.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.Date
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.days
+
+/**
+ * TODO this is a dupe of DateHelper.kt in common module, but it's used in the public module so we can't remove it yet,
+ * nor make it public here because it would be accessible to everyone
+ */
+internal data class DateActiveDupe(val isActive: Boolean, val inGracePeriod: Boolean)
```
Comment (tonidero) [inline public/src/main/java/com/revenuecat/purchases/utils/DateHelperDupe.kt:11]: We should remember we need to make these internal in the original classes as well once we've joined the other modules into one

File: public/src/main/java/com/revenuecat/purchases/utils/Iso8601UtilsDupe.kt [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+     *
+     * @param date ISO string to parse in the appropriate format.
+     * @return the parsed date
+     */
+    fun parse(date: String): Date {
+        return try {
+            var offset = 0
+
+            // extract year
+            val year = parseInt(date, offset, 4.let { offset += it; offset })
```
Comment (tonidero) [inline public/src/main/java/com/revenuecat/purchases/utils/Iso8601UtilsDupe.kt:61]: This is probably not the prettiest code... I've been thinking about simply rewriting this (note that once we join the modules we still need to make this internal), but maybe we can do that in a separate PR

**Review by tonidero — COMMENTED** [review]
Tests seem to be failing? Other than that, this looks good

**Review by tonidero — APPROVED** [review]

**Comment by vegaro** [comment]
I believe the diff in the code coverage is due to the duplicated classes so I am going to merge anyways
